src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -417,16 +402,14 @@ private async Task<bool> TryDeserializeProjectDiagnosticsAsync(IPersistentStorag
                 return false;
             }
 
-            private ValueTask<ImmutableArray<DiagnosticData>> DeserializeDiagnosticsAsync(IPersistentStorageService persistentService, DiagnosticDataSerializer serializer, Project project, TextDocument? document, object key, string stateKey, CancellationToken cancellationToken)
+            private ImmutableArray<DiagnosticData> GetDiagnosticsFromInMemoryStorage(
+                VersionStamp serializerVersion, Project project, TextDocument? document, object key, string stateKey)
             {
                 Contract.ThrowIfFalse(document == null || document.Project == project);
 
-                if (InMemoryStorage.TryGetValue(_owner.Analyzer, (key, stateKey), out var entry) && serializer.Version == entry.Version)
-                {
-                    return ValueTaskFactory.FromResult(entry.Diagnostics);
-                }
-
-                return serializer.DeserializeAsync(persistentService, project, document, stateKey, cancellationToken);
+                return InMemoryStorage.TryGetValue(_owner.Analyzer, (key, stateKey), out var entry) && serializerVersion == entry.Version
+                    ? entry.Diagnostics
+                    : default;
             }
 
             private void RemoveInMemoryCache(DiagnosticAnalysisResult lastResult)
```

Comment:
Is this method still used?

Reply:
yes.  effectively as we produce new results, this is used to purge the previous values we had associated with that analyzer.

This is the code in question:

```c#
            public void SaveToInMemoryStorage(Project project, DiagnosticAnalysisResult result)
            {
                Contract.ThrowIfTrue(result.IsAggregatedForm);
                Contract.ThrowIfNull(result.DocumentIds);

                RemoveInMemoryCache(_lastResult);

                // save last aggregated form of analysis result
                _lastResult = result.ToAggregatedForm();
```